pyerrors/obs.py

```diff
@@ -1069,7 +1069,7 @@ def derived_observable(func, data, array_mode=False, **kwargs):
     for o in raveled_data:
         for name in o.cov_names:
             if name in allcov:
-                if not np.array_equal(allcov[name], o.covobs[name].cov):
+                if not np.allclose(allcov[name], o.covobs[name].cov, rtol=1e-14, atol=1e-14):
```

Comment:
Is the tolerance maybe too small and could cause problems with rounding errors?

Reply:
Good question. I got problems because exact equality was needed before. Maybe rtol=1e-14 is enough to capture differences that are based on finite machine precision? Or even less?
I was thinking about small numbers, but a small atol could lead to problems for entries that are significantly larger than one.

Reply:
I usually use the standard parameters for `allclose` (`rtol=1e-05`, `atol=1e-08`) based on the assumption that this agreement is sufficient for our purposes. When the tolerance approaches machine precision I feel like there could be special cases which cause problems. But this is just a feeling, I never encountered such a case.

Reply:
You are right, any explicit change of the covariance matrix would already be captured using the less strict bounds. Probably it's fine.